File: Result/java/com/xlsxtools/MsgMsgConfigTable.java
package com.xlsxtools;
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class MsgMsgConfigTable
{
	public class MsgMsgConfig
	{
		public	int		id;	//id
		public	String		title;	//标题
		public	String		content;	//内容
		public	MsgMsgSellConfig[]	sell;
		public	int[]		instance;	//副本1
		public	String		remark;	//备注
	}

	public class MsgMsgSellConfig
	{
		public int	id;	//出售获得物品1
		public int	num;	//出售获得物品数量1
	}
	
	private Map<Integer, MsgMsgConfig> m_configs = new HashMap<Integer,MsgMsgConfig>();
	public Map<Integer, MsgMsgConfig> getConfigs()
	{
		return m_configs;
	}

	public void load() throws IOException
	{
		String path = this.getClass().getClassLoader().getResource("Config/Msg_Msg.txt").getPath();
		CSVReader reader = new CSVReader(path, 3, "#");
		while (reader.next()) {
			int index = 0;
			MsgMsgConfig rec = new MsgMsgConfig();

			rec.id = reader.getInt(index++);	//id
			rec.title = reader.getString(index++);	//标题
			rec.content = reader.getString(index++);	//内容
			rec.sell = new MsgMsgSellConfig[3];
			for (int i = 0;i < 3;i ++) {
				rec.sell[i] = new MsgMsgSellConfig();
			}
			rec.sell[0].id = reader.getInt(index++);	//出售获得物品1
			rec.sell[0].num = reader.getInt(index++);	//出售获得物品数量1
			rec.sell[1].id = reader.getInt(index++);	//出售获得物品2
			rec.sell[1].num = reader.getInt(index++);	//出售获得物品数量2
			rec.sell[2].id = reader.getInt(index++);	//出售获得物品3
			rec.sell[2].num = reader.getInt(index++);	//出售获得物品数量3
			rec.instance = new int[4];
			rec.instance[0] = reader.getInt(index++);	//副本1
			rec.instance[1] = reader.getInt(index++);	//副本1
			rec.instance[2] = reader.getInt(index++);	//副本1
			rec.instance[3] = reader.getInt(index++);	//副本1
			rec.remark = reader.getString(index++);	//备注
			m_configs.put(rec.id, rec);
		}
	}
}

File: misc/demo/cpp/xlsxtools/Config/MsgMsgConfigTable.cpp
#include "stdafx.h"
#include "MsgMsgConfigTable.h"
#include "CSVReader.h"


MsgMsgConfigTable::MsgMsgConfigTable()
{
}


MsgMsgConfigTable::~MsgMsgConfigTable()
{
}


map<int, MsgMsgConfig> & MsgMsgConfigTable::GetConfigs()
{
	return m_configs;
}

MsgMsgConfig * MsgMsgConfigTable::GetConfigById(int cid)
{
	if (m_configs.find(cid) != m_configs.end())
	{
		return &m_configs[cid];
	}
	return NULL;
}


bool MsgMsgConfigTable::ConfigProcess(vector<string> fields, MsgMsgConfig & rec)
{
	if (fields.size() < 14)
	{
		return false;
	}
	int idx = 0;

	//错误码
	rec.id = stoi(fields[idx++]);

	//标题
	rec.title = fields[idx++];	

	rec.content = fields[idx++];	//内容

	for (int i = 0; i < 3; i++) {

		rec.sell.push_back(MsgMsgSellConfig());
	}

	
	rec.sell[0].id = stoi(fields[idx++]);	//出售获得物品1

	rec.sell[0].num = stoi(fields[idx++]);	//出售获得物品数量1

	rec.sell[1].id = stoi(fields[idx++]);	//出售获得物品2

	rec.sell[1].num = stoi(fields[idx++]);	//出售获得物品数量2

	rec.sell[2].id = stoi(fields[idx++]);	//出售获得物品3

	rec.sell[2].num = stoi(fields[idx++]);	//出售获得物品数量3

	rec.instance.push_back(stoi(fields[idx++]));	//副本1

	rec.instance.push_back(stoi(fields[idx++]));	//副本1

	rec.instance.push_back(stoi(fields[idx++]));	//副本1

	rec.instance.push_back(stoi(fields[idx++]));	//副本1

	rec.remark = fields[idx++];	//备注

	return true;
}

void MsgMsgConfigTable::Load()
{
	CSVReader reader;
	reader.LoadText("Config/Msg_Msg.txt", 3);
	int rows = reader.GetRowCount();
	for (int r = 0; r < rows; ++r)
	{
		vector<string> row = reader.GetRow(r);
		MsgMsgConfig ac;
		ConfigProcess(row, ac);
		m_configs.insert(std::make_pair(ac.id, ac));
	}
}